field/src/helpers.rs

```diff
@@ -194,3 +216,215 @@ where
 {
     li.zip(ri).map(|(l, r)| l * r).sum()
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
You should implement field-related tests in the `field-testing` crate so that they are called for each of our provided fields. That will allow you to remove the dummy `TestField` struct and its implementation.